### PR TITLE
Pe removemhfe

### PR DESCRIPTION
**Pull request by Paulie14**
Suggesting partial merge of PE_mhfevalues.

Get rid of MH_FEValues class.
- add FE and mapping objects for velocity into DarcyFlowMH
- replace MH_FEValues computation with FEValues
- assemble local_mh_matrix templated by dimension
- replace MH_FEValues in DarcyFlowOutput

DoFHandler and FEValues:
- add function get_loc_dof_indices() to get local indices
- add some getter for FEValues

### CODE REVIEW

File: src/fem/dofhandler.cc [hunk 1]
```diff
@@ -363,6 +363,26 @@ void DOFHandlerMultiDim::get_dof_indices(const CellIterator &cell, unsigned int
...
+        break;
+    case 2:
+        for (unsigned int k=0; k<fe2d_->n_object_dofs(dim,DOF_SINGLE); k++)
+            indices[k] = object_dofs[cell.index()][dim][k] - loffset_;
+        break;
+    case 3:
+        for (unsigned int k=0; k<fe3d_->n_object_dofs(dim,DOF_SINGLE); k++)
+            indices[k] = object_dofs[cell.index()][dim][k] - loffset_;
+        break;
+    }
```
Comment (jstebel) [inline src/fem/dofhandler.cc]: Hint to simplify code: In the switch, set a variable n_object_dofs, and then put the for-cycle out of the switch.

File: src/fem/fe_values.cc [hunk 2]
```diff
@@ -216,7 +216,17 @@ const Quadrature<dim> * FEValuesBase<dim,spacedim>::get_quadrature() const
...
+const FiniteElement<dim,spacedim> * FEValuesBase<dim,spacedim>::get_fe() const
+{
+    return fe;
+}
 
+template<unsigned int dim, unsigned int spacedim> inline
+const Mapping<dim,spacedim> * FEValuesBase<dim,spacedim>::get_mapping() const
+{
+    return mapping;
+}
```
Comment (jstebel) [inline src/fem/fe_values.cc]: "inline" makes sense only in header file.

Comment (jbrezmorf) [inline src/fem/fe_values.cc]: So we should move implementation of critical getters into the header file.

File: src/flow/darcy_flow_mh_output.cc [hunk 3]
```diff
@@ -243,29 +245,58 @@ void DarcyFlowMHOutput::make_element_scalar() {
...
+                              * fv_rt2.shape_vector(li,0); //fe_values.RT0_value( ele, ele->centre(), li )
+                }
+                break;
+            case 3: 
+                fv_rt3.reinit(ele);
+                for (unsigned int li = 0; li < ele->n_sides(); li++) {
+                    flux_in_center += mhdh.side_flux( *(ele->side( li ) ) )
+                              * fv_rt3.shape_vector(li,0); //fe_values.RT0_value( ele, ele->centre(), li )
+                }
+                break;
```
Comment (jbrezmorf) [inline src/flow/darcy_flow_mh_output.cc]: This and other code involving different dimensions is terribly redundant. 
I suggest create an internal class template DimensionData<dim>, move all dimension dependent data members there
and move parts of assembly routine there.

**Pull request merged**